python/src/scipp/neutron/instrument_view.py

```diff
@@ -24,15 +22,13 @@ def _rebin_histogram_data(obj, bins):
     new_bins = bins
     if not isinstance(bins, dict):
         new_bins = {"tof": bins}
-    if not sc.contains_events(obj):
-        for dim, val in new_bins.items():
-            obj = sc.rebin(obj, dim,
-                           su.make_bins(data_array=obj, bins=val, dim=dim))
-        # Once data has been histogrammed, we return None as the new bins,
-        # since the plot function expects event data if bins are specified.
-        return obj, None
-    else:
-        return obj, new_bins
+    for dim, val in new_bins.items():
+        obj = sc.rebin(obj, dim, su.make_bins(data_array=obj,
+                                              bins=val,
+                                              dim=dim))
+    # Once data has been histogrammed, we return None as the new bins,
+    # since the plot function expects event data if bins are specified.
+    return obj, None
```

Comment:
@nvaytet Is this the right approach, just remove the old events branch everywhere in the plotting code? I think we will anyway need to make additions so plotting can histogram bucketed data (as a follow-up step)?

Reply:
Yes fine for now, see below